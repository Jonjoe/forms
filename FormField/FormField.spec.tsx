import React from "react";
import renderer from "react-test-renderer";

import {
  buildLabelClassNames,
  buildInputClassNames,
  // processValidators,
} from "./FormField.component";

import Example from "./FormField.example";

test("FormField renders correctly", () => {
  const component = renderer.create(<Example />);

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

describe("FormField functions", () => {
  describe("buildLabelClassNames()", () => {
    test("if the field is untouched", () => {
      const expectedResult = "ClassName__Label";
      const actualResult = buildLabelClassNames(
        "ClassName",
        false,
        false,
        false
      );

      expect(actualResult).toBe(expectedResult);
    });

    test("if the field has a value", () => {
      const expectedResult = "ClassName__Label ClassName__Label--containsValue";
      const actualResult = buildLabelClassNames(
        "ClassName",
        true,
        false,
        false
      );

      expect(actualResult).toBe(expectedResult);
    });

    test("if the field contains an error", () => {
      const expectedResult = "ClassName__Label ClassName__Label--containsError";
      const actualResult = buildLabelClassNames(
        "ClassName",
        false,
        true,
        false
      );

      expect(actualResult).toBe(expectedResult);
    });

    test("if the field has been touched", () => {
      const expectedResult = "ClassName__Label ClassName__Label--isDirty";
      const actualResult = buildLabelClassNames(
        "ClassName",
        false,
        false,
        true
      );

      expect(actualResult).toBe(expectedResult);
    });

    test("if all the conditions are met", () => {
      const expectedResult =
        "ClassName__Label ClassName__Label--containsValue ClassName__Label--containsError ClassName__Label--isDirty";
      const actualResult = buildLabelClassNames("ClassName", true, true, true);

      expect(actualResult).toBe(expectedResult);
    });

    test("if all but one of the conditions are met", () => {
      const expectedResult =
        "ClassName__Label ClassName__Label--containsValue ClassName__Label--isDirty";
      const actualResult = buildLabelClassNames("ClassName", true, false, true);

      expect(actualResult).toBe(expectedResult);
    });
  });

  describe("buildInputClassNames()", () => {
    test("if the field is untouched", () => {
      const expectedResult = "ClassName__Input";
      const actualResult = buildInputClassNames("ClassName", false);

      expect(actualResult).toBe(expectedResult);
    });

    test("if the field contains an error", () => {
      const expectedResult = "ClassName__Input ClassName__Input--containsError";
      const actualResult = buildInputClassNames("ClassName", true);

      expect(actualResult).toBe(expectedResult);
    });
  });
});
