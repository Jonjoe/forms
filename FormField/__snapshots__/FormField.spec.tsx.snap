// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders correctly 1`] = `
<fieldset
  className="blockClassName false"
>
  <label
    className="Label FormField__Label"
    htmlFor="Example Label"
  >
    Example Label
  </label>
  <input
    className="SingleLineField FormField__Input"
    id="Example Label"
    onChange={[Function]}
    value=""
  />
  <div />
</fieldset>
`;
